include/networkit/algebraic/algorithms/AlgebraicSpanningEdgeCentrality.hpp

```diff
@@ -82,34 +83,35 @@ void AlgebraicSpanningEdgeCentrality<Matrix>::runApproximation() {
     scoreData.resize(m, 0.0);
     double epsilon2 = tol * tol;
     const count k = ceil(log2(n)) / epsilon2;
-    double randTab[3] = {1.0/sqrt(k), -1.0/sqrt(k)};
+    double randTab[2] = {1.0/sqrt(k), -1.0/sqrt(k)};
 
-    std::vector<Vector> yRows(k, Vector(n));
-    std::vector<Vector> zRows(k, Vector(n));
-
-    G.forEdges([&](node u, node v, edgeweight w, index) {
-#pragma omp parallel for
-        for (omp_index i = 0; i < static_cast<omp_index>(k); ++i) {
+    const auto rhsLoader = [&](count i, Vector& yRow) {
+        yRow.fill(0);
+        G.forEdges([&](node u, node v, edgeweight w, index) {
             double rand = randTab[Aux::Random::integer(1)];
-            yRows[i][u] += w * rand;
-            yRows[i][v] -= w * rand;
-        }
-    });
-
+            yRow[u] += w * rand;
+            yRow[v] -= w * rand;
+        });
+        return yRow;
+    };
+
+    std::vector<std::atomic<double>> scoreDataAtom(m);
+    const auto resultProcessor = [&](count i, const Vector& zRow) {
+        G.forEdges([&](node u, node v, edgeid e) {
+            double diff = (zRow[u] - zRow[v]);
+            diff *= diff;
+            for (double old = scoreDataAtom[e]; !scoreDataAtom[e].compare_exchange_strong(old, old + diff););
```

Comment:
not sure about this

Reply:
Can't we use `fetch_add`? Alternatively, it might be clear to use a matrix that is indexed by `(thread ID) x node` such that we can aggregate over all threads after finishing the computation (and thus we can drop the atomics here).

Reply:
sadly std::atomic<double>::fetch_add is specialized only in c++20. Sure using thread matrix would be much more intuitive, but I think it might be worth sacrificing readability here in favor of saving memory

Reply:
I thought again, and it will actually not be very intuitive, the profit will only be in getting rid of the atomics

Reply:
Okay, but if we keep the atomics, we can at least reduce their memory ordering requirements and pass `std::memory_order_relaxed` to the `compare_exchange_strong` and the atomic load in the loop that sets `scoreData` below.

After this change is done, the PR can be merged.